docs/4.4/api-reference.md

```diff
@@ -11,57 +11,287 @@ Teleport is currently working on documenting our API.
+
+## GoLang Examples
+Below are some code examples that can be used with Teleport to perform a few key tasks.
+
+Before you begin: 
+
+- Install [Golang](https://golang.org/doc/install) 1.13+ and Setup Go Dev Environment
+- Have access to a Teleport Auth server ([quickstart](quickstart.md))
+- [go get](https://golang.org/pkg/cmd/go/internal/get) the following packages (as you need them)
+    - github.com/gravitational/teleport
+    - github.com/gravitational/teleport/lib/auth
+  	- github.com/gravitational/teleport/lib/utils
+    - github.com/gravitational/teleport/lib/services
+
+## Authentication 
+
+In order to interact with the API, you will need to provision appropriate
 TLS certificates. In order to provision certificates, you will need to create a
-user with appropriate permissions:
-
-```bash
-$ cat > rscs.yaml <<EOF
-kind: user
-metadata:
-  name: access-plugin
-spec:
-  roles: ['access-plugin']
-version: v2
----
-kind: role
-metadata:
-  name: access-plugin
-spec:
-  allow:
-    rules:
-      - resources: ['access_request']
-        verbs: ['list','read','update']
-    # teleport currently refuses to issue certs for a user with 0 logins,
-    # this restriction may be lifted in future versions.
-    logins: ['access-plugin']
-version: v3
+user with appropriate permissions. You should only give the api user permissions for what it actually needs. 
+
+To quickly get started with the api, you can use this api-admin user, but in real usage make sure to have stringent permissions in place.
+
+```yaml
+# Copy and Paste the below and run on the Teleport Auth server.
+$ cat > api-admin.yaml <<EOF
+{!examples/go-client/api-admin.yaml!}
 EOF
-# ...
-$ tctl create rscs.yaml
-# ...
-$ tctl auth sign --format=tls --user=access-plugin --out=auth
-# ...
+
+$ tctl create -f api-admin.yaml
+$ mkdir -p certs
+$ tctl auth sign --format=tls --user=api-admin --out=certs/api-admin 
 ```
 
-The above sequence should result in three PEM encoded files being generated:
-`auth.crt`, `auth.key`, and `auth.cas` (certificate, private key, and CA certs respectively).
+This should result in three PEM encoded files being generated in the `/certs` directory: `api-admin.crt`, `api-admin.key`, and `api-admin.cas` (certificate, private key, and CA certs respectively).
 
-Note: by default, tctl auth sign produces certificates with a relatively short lifetime.
-For production deployments, the --ttl flag can be used to ensure a more practical
-certificate lifetime.
+!!! Note
+    By default, `tctl auth sign` produces certificates with a relatively short lifetime.
+    For production deployments, the `--ttl` flag can be used to ensure a more practical
+    certificate lifetime.
```

Comment:
It would be good to provide an example of simple system / weekly job that would be able to obtain new certs for users.

Reply:
Yeah that's a good idea. I'm not sure of the best way to do this though, i.e. how we would recommend doing this to users.

Do you want me to make an automated process for the auth server to sign the certificates and then send the PEM files to wherever the API is running? How would the PEM files be transferred?

Or did you want me to use the API itself to regenerate the certificates? I don't see a way to do this easily either, as most of the logic for `auth sign` is baked into `tctl` rather than the API.

It looks like nodes in a cluster will automatically request new certificates when their certs enter the rotation grace period, would something similar to this be possible?

Reply:
This is a great question. The proper solution is to have this built into teleport, but this is out of scope of this. You can recommend to create a periodic job using Systemd or Kubernetes to rotate certificates on schedule though. We will provide a better solution for this problem later though.

Reply:
+1 klizhentas comment.  let's just leave a small comment for now and we'll come back to this later.